### DISCUSSION TITLE
Files ownership user vs root

### DISCUSSION THREAD

**Discussion by TacoV** [dunglas/symfony-docker#296] [Q&A] [answered]
When I start a project from scratch, under the user `tacov`, the files generated from within the containers are owned by `root`.
I "fix" this by `chown`ing from time to time, but that does not feel right. Is this a bad setup on my part? Can this be prevented?

```
git clone https://github.com/dunglas/symfony-docker.git
cd symfony-docker
touch composer.tmp symfony.tmp
docker compose build --pull --no-cache
# Up to here, all files are owned by tacov:tacov
docker compose up
rm composer.tmp symfony.tmp
# All new files and folders, such as composer.json, var/, src/, are owned by root:root
```

**Comment by TacoV** [comment]
Found an approach that does not require `sudo` permissions
https://github.com/dunglas/symfony-docker/blob/main/docs/troubleshooting.md#editing-permissions-on-linux:

> If you work on linux and cannot edit some of the project files right after the first installation, you can run `docker compose run --rm php chown -R $(id -u):$(id -g) .` to set yourself as owner of the project files that were created by the docker container.

It does not solve the underlying issue, though.

**Reply by maxhelias** [reply, marked as answer]
The ownership with docker is complicated. See more explanation : https://github.com/dunglas/symfony-docker/issues/325#issuecomment-1304628407
